frontend/views/containers/chatroom/poll-message-content/PollToVote.vue

```diff
@@ -0,0 +1,173 @@
+<template lang='pug'>
+form(@submit.prevent='')
+  i18n.c-poll-label(tag='label') poll
+  h3.is-title-4.c-poll-title {{ pollData.question }}
+
+  fieldset.is-column
+    label.c-poll-option(
+      v-for='option in pollData.options'
+      :key='option.id'
+      :class='allowMultipleChoices ? "checkbox" : "radio"'
+    )
+      input.input(v-if='allowMultipleChoices' type='checkbox' :value='option.id' v-model='form.selectedOptions')
+      input.input(v-else type='radio' :name='messageId' :value='option.id' v-model='form.selectedOptions')
+      span.c-poll-option-value {{ option.value }}
+
+  banner-scoped(ref='errBanner')
+
+  .buttons.c-buttons-container
+    template(v-if='enableSubmitBtn')
+      i18n.is-small(v-if='isChangeMode' tag='button' type='button' @click='changeVotes') Change vote
+      i18n.is-small(v-else tag='button' type='button' @click='submitVotes') Submit
+
+    i18n.is-small.is-outlined(v-if='isChangeMode' tag='button' type='button' @click='onCancelClick') Cancel
+</template>
+
+<script>
+import sbp from '@sbp/sbp'
+import { mapGetters } from 'vuex'
+import { cloneDeep } from '@model/contracts/shared/giLodash.js'
+import { POLL_TYPES } from '@model/contracts/shared/constants.js'
+import BannerScoped from '@components/banners/BannerScoped.vue'
+
+export default ({
+  name: 'PollToVote',
+  inject: ['pollUtils'],
+  components: {
+    BannerScoped
+  },
+  props: {
+    pollData: Object,
+    messageId: String,
+    messageHash: String,
+    isChangeMode: Boolean
+  },
+  data () {
+    const initVal = () => this.pollData.pollType === POLL_TYPES.MULTIPLE_CHOICES ? [] : ''
+
+    return {
+      form: {
+        selectedOptions: initVal()
+      },
+      selectedOptionsBeforeUpdate: initVal()
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'ourUsername',
+      'currentChatRoomId'
+    ]),
+    allowMultipleChoices () {
+      return this.pollData.pollType === POLL_TYPES.MULTIPLE_CHOICES
+    },
+    enableSubmitBtn () {
+      if (this.isChangeMode) {
+        return this.formBeenUpdated()
+      } else {
+        return this.allowMultipleChoices
+          ? this.form.selectedOptions.length > 0
+          : Boolean(this.form.selectedOptions)
+      }
+    },
+    selectedVotesAsString () {
+      return this.allowMultipleChoices
+        ? this.form.selectedOptions.map(id => this.getOptValue(id)).join(', ')
+        : this.getOptValue(this.form.selectedOptions)
+    }
+  },
+  methods: {
+    submitVotes () {
+      try {
+        sbp('gi.actions/chatroom/voteOnPoll', {
```

Comment:
[Forgot this one!](https://github.com/okTurtles/group-income/pull/1628#discussion_r1271373035)